app/serializers/chat_message_serializer.rb

```diff
@@ -13,7 +13,7 @@ class ChatMessageSerializer < ApplicationSerializer
     :edited,
     :reactions
 
-  has_one :user, serializer: BasicUserSerializer, embed: :objects
+  has_one :user, serializer: UserCardSerializer, embed: :objects
```

Comment:
The UserCardSerializer includes a data from a *lot* of tables, so will be much slower. Do we need all of it?

To avoid N+1s, we'll need to make sure all the data is preloaded like

https://github.com/discourse/discourse/blob/b65ecf69871656155c381343cd84de919cbbf803/app/controllers/users_controller.rb#L112-L120

Reply:
Yep this is why I pointed to it in code-review. No we don't need all of it, we just need the flair stuff. But In the end that would mean a custom user serializer no?

Reply:
Yeah either a custom user serializer, or you could wrap up the user in a ChatPosterSerializer (similar to core's TopicPosterSerializer)

https://github.com/discourse/discourse/blob/e599b5b08f4323d50fb2dd584dbb06333d60b995/app/serializers/topic_poster_serializer.rb

Reply:
Will look at this, was looking for something like this, thanks